### DISCUSSION TITLE
Bug in Tasmota 5.2.1: Download of configuration backup file closes connection at 1492 bytes.

### DISCUSSION THREAD

**Discussion by hs06** [arendst/Tasmota#20327] [Support request] [answered]
Dear all, possibly someone has an easy answer to this. As pointed out in the subject: the download of the backup configuration file does not work: The TCP connection is reset after 1492 bytes. This looks like a tcp issue to me. I am downloading from a Mac (MacOS 13.3.1, 22E772610a). The Tasmota device is a "sonoff TH" on version 5.2.1 

`% wget "http://192.168.1.210/dl?"  
--2023-12-28 09:10:12--  http://192.168.1.210/dl?
Connecting to 192.168.1.210:80... connected.
HTTP request sent, awaiting response... No data received.
Retrying.

--2023-12-28 09:10:15--  (try: 2)  http://192.168.1.210/dl?
Connecting to 192.168.1.210:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4096 (4,0K) [application/octet-stream]
Saving to: ‘dl?’

dl?                                                            36%[====================================================>                                                                                             ]   1,46K  --.-KB/s    in 1,9s    

2023-12-28 09:10:17 (780 B/s) - Connection closed at byte 1492. Retrying.

--2023-12-28 09:10:19--  (try: 3)  http://192.168.1.210/dl?
Connecting to 192.168.1.210:80... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4096 (4,0K) [application/octet-stream]
Saving to: ‘dl?’

dl?                                                            36%[====================================================>                                                                                             ]   1,46K  --.-KB/s    in 1,9s    

2023-12-28 09:10:21 (778 B/s) - Connection closed at byte 1492. Retrying.
`

The reason why I am downloading the configuration file: I want to upgrade the Tasmota on the device. It has run for nearly 6 years flawlessly, but now loses WiFi connection frequently (and does not reconnect). There are new WiFi networks in the vicinity (around 30 in total), so the 2.4GHz channels are very very crowded, but I can't do a lot. I was expecting possibly a better reconnect behaviour and could also do my logic on-board if I remember correctly (the Tasmota device controls the warm water of the apartment and is operated by a FHEM).

Any idea appreciated -- otherwise I will just upgrade the device and reconfigure it manually.

A big thanks to @arendst -- I love Tasmota. You have created an amazing (AMAZING!) operating system & ecosystem. It's more than a little firmware, it is a real system indeed. 

All the best and a good holiday season to everyone!
Hendrik

**Comment by barbudor** [comment]
Do you have so many devices that you have script that?
Why not downloading through the menu (although I don't know if this exist in tasmota 5.x)

There is also a tool name decode-config which can do that for you (if v5.x is supported)
It can generate a binary dump of the config as well as a JSON which could be helpful too
See in repo 'tasmota/decode-config'

**Comment by Jason2866** [comment]
Before doing the upgrade dance for all devices, i would try with a test device IF the actual Tasmota version does solve the issue.

**Comment by sfromis** [comment, marked as answer]
Beware that possible reasons for instability of very old devices include hardware degradation over time. With current Tasmota versions, I'm seeing occasional cases of the hardware watchdog biting on old Sonoff TH16 devices from 2018, and I'm not assuming this to be due to an error in Tasmota.

**Comment by hs06** [comment]
Thanks a lot for your suggestions. No: I'm actually not scripting the upgrade. The download also fails in Safari and Chrome -- I used wget to see some details, i.e. that the connection is closed exactly after 1492 bytes have been delivered. Also for every retry the connection is closed.

I will go the upgrade path on the device (a single TH16) tomorrow. Thanks @sfromis for the suggestion of an ageing hardware/fault. I will get a new replacement device just in case.

@barbudor: just to clarify: I can't download the backup configuration, so the decode-config tool (that I have already used on another device) won't help in this case.

**Comment by hs06** [comment]
just to finalize this: 
The update routine by uploading the new image had not worked, via the HTTP URI it worked without problem. I updated to 5.14.0 and was able to backup the configuration.

however, the following update to 6.7.1 did not work. The connection dropped while downloading. I then supplied the -minimal binary, but the device didn’t update. It seems bricked now, so **I assume there is/was a hardware defect**, e.g. of some memory areas.

**Comment by hs06** [comment]
Dear all,
to finalize it once more, I have returned to the device with an a serialUSB device and receive the following output, which looks like a boot-loop to me, possibly caused by the flash-corruption that had been suspected earlier.


 ```
ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v11c3757b
~ld


 ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v11c3757b
~ld


 ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v11c3757b
~ld


 ets Jan  8 2013,rst cause:4, boot mode:(3,7)

wdt reset
load 0x4010f000, len 1384, room 16 
tail 8
chksum 0x2d
csum 0x2d
v11c3757b
~ld
```




I have changed the TH16 for the same type now (.. however, that one worked fine on the USB port but not on 220v - but that's a different story that I have to figure out).

Thanks for the support of everyone on this thread.
